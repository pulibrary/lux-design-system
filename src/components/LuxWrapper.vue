<template>
  <component
    :is="type"
    :class="['lux-wrapper', { 'lux-full-width': fullWidth }, horizontal]"
    :style="{ 'max-width': maxWidth + 'px' }"
  >
    <slot />
  </component>
</template>

<script>
/**
 * Used to build the outer wrapper of a page, including the page title and
 * associated actions. Wrapper can be span the full width of the viewport or limited to a max-width of 1440px.
 */
export default {
  name: "LuxWrapper",
  status: "ready",
  release: "1.0.0",
  type: "Element",
  props: {
    /**
     * The html element name used for the wrapper.
     */
    type: {
      type: String,
      default: "div",
    },
    /**
     * The maximum width of the wrapper. Default is set to 1170.
     */
    maxWidth: {
      type: [String, Number],
      default: 1170,
    },
    /**
     * Determines whether the wrapper takes up 100% of the parent container.
     */
    fullWidth: {
      type: [String, Boolean],
      default: false,
    },
    /**
     * Determines how the flexbox wrapper is horizontally aligned. This value defaults to
     * having space between grid items. Options include `start`, `center`, and `end`.
     */
    horizontal: {
      type: String,
      default: "",
    },
  },
}
</script>

<style lang="scss" scoped>
.lux-wrapper {
  margin: auto;

  &.lux-full-width {
    padding: 0;
    max-width: 100%;
    width: 100%;
  }
}
</style>

<docs>
  ```jsx
    <div>
      <lux-wrapper>Wrapper can be used to wrap any components together.</lux-wrapper>
      <lux-wrapper :max-width="1400">Wrapper can be used to wrap any components together.</lux-wrapper>
    </div>
  ```
</docs>
