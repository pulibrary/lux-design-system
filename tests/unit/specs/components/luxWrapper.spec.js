import { mount } from "@vue/test-utils"
import LuxWrapper from "@/components/LuxWrapper.vue"
import { nextTick } from "vue"

describe("LuxWrapper.vue", () => {
  let wrapper

  it("can accept an arbitrary number of pixels in the maxWidth prop", () => {
    wrapper = mount(LuxWrapper, {
      props: {
        maxWidth: 123,
      },
    })
    expect(wrapper.find("div").attributes("style")).toEqual("max-width: 123px;")
  })

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
