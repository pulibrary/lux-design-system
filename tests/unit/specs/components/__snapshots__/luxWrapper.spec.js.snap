// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LuxWrapper.vue has the expected html structure 1`] = `
<div
  class="lux-wrapper"
  style="max-width: 123px;"
>
  
  
</div>
`;
